README.md

```diff
@@ -1,27 +1,395 @@
-# To devs
+## Generic Bucket Provisioning
 
-Clone this repo with 
+Kubernetes natively supports dynamic provisioning for many types of file and
+block storage, but lacks support for object bucket provisioning. 
+This repo is a temporary placeholder for an object store bucket provisioning library,
+very similar to the Kubernetes 
+[sig-storage-lib-external-provisioner](https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/master/controller/controller.go)
+library. The goal is to eventually move this repo to a Kubernetes repo within _sig-storage/_.
 
-`go get -d github.com/yard-turkey/lib-bucket-provisioner`
+### Design
+The time has come where we can support a bucket provisioning API similar to that used for
+Persistent Volumes. We propose two new Custom Resources to abstract an object store bucket
+and a claim/request for such a bucket.  It's important to keep in mind that this proposal 
+only defines bucket and bucket claim APIs and related library code. The lib ensures that 
+the _contract_ made to app developers regarding the artifacts of bucket creation (Secrets,
+ConfigMaps) are guaranteed. The actual creation of buckets belongs to each object store
+provisioner. However, as is true for dynamic Persistent Volume provisioning, each bucket
+provisioner needs to adhere to the inferfaces defined in the bucket library.
 
-Then install the dependencies
+An `ObjectBucketClaim` (OBC) is similar in usage to a Persistent Volume Claim
+and an `ObjectBucket` (OB) is the Persistent Volume equivalent. 
+Bucket binding refers to the actual bucket being created by the underlying object
+store provide, and the generation of artifacts which will be consumed by application pods.
+An OBC is namespaced and references a storage class which defines the object store. The
+details of the object store (ceph, minio, cloud, on-prem) are not visible to the app pod
+and can change without disturbing the app in anyway. An OB is non-namespaced (global),
```

Comment:
> can change without disturbing the app in anyway.

This is maybe a little unclear.  It reads to me as if the object store could be hot-swapped during the app's runtime.